bazel/ray_deps_setup.bzl

```diff
@@ -1,128 +1,198 @@
 load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository", "new_git_repository")
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
-def ray_deps_setup():
-    RULES_JVM_EXTERNAL_TAG = "2.10"
-    RULES_JVM_EXTERNAL_SHA = "1bbf2e48d07686707dd85357e9a94da775e1dbd7c464272b3664283c9c716d26"
+def github_repository(**kwargs):
+    """
+    Conveniently chooses between ZIP, git, etc. GitHub repositories.
+    We prefer ZIP downloads, since they're smaller and faster
+    due to the lack of history.
+
+    The remote parameter is required.
+    One of {commit, tag, branch} is also required.
+    The name parameter is optional but recommended.
+    Set sha256 = True if the ZIP archive hash is unknown, then update it ASAP
+    to allow caching & avoid repeated downloads on every build.
+
+    If build_file is a string, it refers to a custom build file.
+    If build_file is absent, the project is assumed to contain a build file.
+    If build_file == True, then the build file name is auto-deduced.
+
+    If path   is present , local repository is assumed.
+    If sha256 == <string>, obtains ZIP download with the given hash (fast)
+    If sha256 == True    , obtains ZIP download without knowing the hash (fast)
```

Comment:
Yes, as follows:

    Set sha256 = True if the ZIP archive hash is unknown, then update it ASAP
    to allow caching & avoid repeated downloads on every build.

Reply:
Is it used by any of our build targets though? No point in adding it if not and this seems like an easy way to accidentally end up with slow builds.

Reply:
No what I mean is that it should never be checked in (similar to how `local` will be checked in), but it is used locally to download the ZIP without knowing the hash a priori. And as a bonus Bazel tells you what the hash is when you do that, so it's quite easy to just copy-paste from the terminal that's already in front of you. Otherwise you have force everyone (including yourself) to manually download the ZIP in the browser and run `sha256sum` before you can add a dependency at all. And then you have to have Bazel download it one _more_ time now that it has the hash. Which is both quite painful in itself, and also a time-waster given that some ZIPs are fairly large and now have to be downloaded twice.